src/main/resources/lib/ssb/dataset/calculator.ts

```diff
@@ -53,22 +53,38 @@ export function getKpiDatasetMonth(config: Content<CalculatorConfig>): Dataset |
   return kpiDatasetMonthRepo ? JSONstat(kpiDatasetMonthRepo.data).Dataset('dataset') : null
 }
 
-export function getPifDataset(config: Content<CalculatorConfig>): Dataset | null {
-  const pifSource: Content<GenericDataImport & DataSource> | null = config?.data.pifSource
+export const PIF_CALCULATOR = 'pifCalculator'
+export const BPI_CALCULATOR = 'bpiCalculator'
+// TODO: We could make a generic one for calculators with multiple datasources, such as kpi and bkibol
+export function getCalculatorDatasetFromSource(
+  config: Content<CalculatorConfig>,
+  calculator: typeof PIF_CALCULATOR | typeof BPI_CALCULATOR
+): Dataset | null {
+  let dataSourceConfig = null
+
+  if (calculator === PIF_CALCULATOR) {
+    dataSourceConfig = config?.data.pifSource
+  }
+
+  if (calculator === BPI_CALCULATOR) {
+    dataSourceConfig = config?.data.bpiSource
+  }
+
+  const dataSource: Content<GenericDataImport & DataSource> | null = dataSourceConfig
     ? getContent({
-        key: config.data.pifSource,
+        key: dataSourceConfig,
       })
     : null
 
-  if (pifSource === null) {
-    log.info('Data calculator - pifSource is Null, calculatorConfig: ' + JSON.stringify(config, null, 4))
+  if (dataSource === null) {
+    log.info(`Data calculator - ${calculator} source is Null, calculatorConfig: ` + JSON.stringify(config, null, 4))
   }
 
-  const pifDatasetRepo: DatasetRepoNode<JSONstatType> | null = pifSource
-    ? (datasetOrUndefined(pifSource) as DatasetRepoNode<JSONstatType> | null)
+  const datasetRepo: DatasetRepoNode<JSONstatType> | null = dataSource
+    ? (datasetOrUndefined(dataSource) as DatasetRepoNode<JSONstatType> | null)
     : null
 
-  return pifDatasetRepo ? JSONstat(pifDatasetRepo.data).Dataset('dataset') : null
+  return datasetRepo ? JSONstat(datasetRepo.data).Dataset('dataset') : null
 }
 
```

Comment:
Må legge til ny kalkulator i getAllCalculatorDataset() slik at den oppdateres sammen med de andre :)

Reply:
Jeg skal oppdatere funksjonen men vi burde vente med å implementere det for de andre kalkulatorene ☺️ Jeg kommer til å notere meg det når jeg oppretter refaktoreringssakene for hver kalkulator